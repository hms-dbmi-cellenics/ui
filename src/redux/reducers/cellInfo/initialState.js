const initialState = {
  component: null,
  cellId: null,
  geneName: null,
  trackCluster: null,
  focus: {
    store: 'cellSets',
    key: 'louvain',
  },
  groupedTrack: 'louvain',
  selectedTracks: ['louvain'],
};

export default initialState;
