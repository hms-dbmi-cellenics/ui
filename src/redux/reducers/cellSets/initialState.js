const initialState = {
  properties: {},
  hierarchy: [],
  selected: {},
  initialLoadPending: true,
  loading: false,
  error: false,
  updatingClustering: false,
  hidden: new Set(),
};

export default initialState;
