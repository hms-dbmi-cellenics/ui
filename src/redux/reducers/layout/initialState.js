const initialLayoutSpatial = {
  windows: {
    direction: 'row',
    first: {
      first: {
        first: 'UMAP', second: 'Spatial', direction: 'row', splitPercentage: 62,
      },
      second: {
        first: 'Heatmap',
        second: 'Cell sets and Metadata',
        direction: 'row',
        splitPercentage: 62,
      },
      direction: 'column',
      splitPercentage: 45,
    },
    second: 'Genes',
    splitPercentage: 70,
  },
  panel: 'Gene list',
};

const initialLayoutSingleCell = {
  windows: {
    direction: 'row',
    first: {
      first: {
        first: 'UMAP', second: 'Cell sets and Metadata', direction: 'row', splitPercentage: 62,
      },
      second: 'Heatmap',
      direction: 'column',
      splitPercentage: 45,
    },
    second: 'Genes',
    splitPercentage: 70,
  },
  panel: 'Gene list',
};

export { initialLayoutSpatial, initialLayoutSingleCell };
