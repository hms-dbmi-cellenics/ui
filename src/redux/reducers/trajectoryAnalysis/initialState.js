const initialState = {
  loading: false,
  error: false,
  plotData: {},
};

export default initialState;
