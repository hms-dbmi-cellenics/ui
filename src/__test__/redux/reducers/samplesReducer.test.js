import samplesReducer from '../../../redux/reducers/samples';
import initialState, { sampleTemplate, sampleFileTemplate } from '../../../redux/reducers/samples/initialState';

import {
  SAMPLES_CREATE,
  SAMPLES_UPDATE,
  SAMPLES_FILE_UPDATE,
  SAMPLES_LOADED,
  SAMPLES_ERROR,
  SAMPLES_DELETE,
  SAMPLES_SAVING,
  SAMPLES_SAVED,
  SAMPLES_METADATA_DELETE,
} from '../../../redux/actionTypes/samples';

describe('samplesReducer', () => {
  const mockUuid1 = 'asd123';
  const mockUuid2 = 'qwe234';
  const fileName = 'features.tsv';

  const sample1 = {
    ...sampleTemplate,
    name: 'test sample',
    uuid: mockUuid1,
    createdDate: '2021-01-01T14:48:00.000Z',
    lastModified: '2021-01-01T14:48:00.000Z',
  };

  const sample2 = {
    ...sampleTemplate,
    name: 'test sample 2',
    uuid: mockUuid2,
    createdDate: '2021-01-02T14:48:00.000Z',
    lastModified: '2021-01-02T14:48:00.000Z',
  };

  const updateActionResult = {
    ...sample1,
    name: 'updated name',
  };

  const oneSampleState = {
    ...initialState,
    [sample1.uuid]: sample1,
  };

  const twoSamplesState = {
    ...oneSampleState,
    [sample2.uuid]: sample2,
  };

  const mockFile = {
    ...sampleFileTemplate,
    name: fileName,
  };

  it('Reduces identical state on unknown action', () => expect(
    samplesReducer(undefined, {
      action: 'well/this/is/not/a/valid/action',
      payload: {},
    }),
  ).toEqual(initialState));

  it('Inserts a new sample correctly', () => {
    const newState = samplesReducer(initialState, {
      type: SAMPLES_CREATE,
      payload: {
        sample: sample1,
      },
    });

    expect(newState[sample1.uuid]).toEqual(sample1);
    expect(newState).toMatchSnapshot();
  });

  it('Adds a new sample correctly', () => {
    const newState = samplesReducer(oneSampleState, {
      type: SAMPLES_CREATE,
      payload: {
        sample: sample2,
      },
    });

    expect(newState[sample1.uuid]).toEqual(sample1);
    expect(newState[sample2.uuid]).toEqual(sample2);
    expect(newState).toMatchSnapshot();
  });

  it('Updates a sample correctly', () => {
    const newState = samplesReducer(oneSampleState, {
      type: SAMPLES_UPDATE,
      payload: {
        sampleUuid: mockUuid1,
        sample: updateActionResult,
      },
    });

    expect(newState[sample1.uuid]).toEqual(updateActionResult);
    expect(newState).toMatchSnapshot();
  });

  it('Updates sample files correctly', () => {
    const newState = samplesReducer(oneSampleState, {
      type: SAMPLES_FILE_UPDATE,
      payload: {
        sampleUuid: mockUuid1,
        fileName,
        fileDiff: mockFile,
        lastModified: 'newLastModified',
      },
    });

    expect(newState[sample1.uuid].fileNames).toEqual([fileName]);
    expect(newState[sample1.uuid].files[fileName]).toEqual(mockFile);
    expect(newState).toMatchSnapshot();
  });

  it('Delete samples correctly', () => {
    const newState = samplesReducer(twoSamplesState, {
      type: SAMPLES_DELETE,
      payload: {
        sampleUuids: [sample2.uuid],
      },
    });

    expect(newState[sample2.uuid]).toBeUndefined();
    expect(newState).toMatchSnapshot();
  });

  it('Updates sample files correctly', () => {
    const newState = samplesReducer(oneSampleState, {
      type: SAMPLES_FILE_UPDATE,
      payload: {
        sampleUuid: mockUuid1,
        fileName,
        fileDiff: mockFile,
        lastModified: 'newLastModified',
      },
    });

    expect(newState[sample1.uuid].fileNames).toEqual([fileName]);
    expect(newState[sample1.uuid].files[fileName]).toEqual(mockFile);
    expect(newState).toMatchSnapshot();
  });

  it('Loads samples correctly', () => {
    const newState = samplesReducer(initialState, {
      type: SAMPLES_LOADED,
      payload: {
        samples: {
          ids: [sample1.uuid, sample2.uuid],
          [sample1.uuid]: sample1,

          [sample2.uuid]: sample2,
        },
      },
    });

    expect(newState.meta.loading).toEqual(false);
    expect(newState.meta.error).toEqual(false);
    expect(newState).toMatchSnapshot();
  });

  it('Sets up saving state correctly', () => {
    const savingMsg = 'Saving';

    const newState = samplesReducer({
      ...oneSampleState,
      meta: {
        ...oneSampleState[sample1.uuid].meta,
        loading: false,
        saving: false,
        error: false,
      },
    }, {
      type: SAMPLES_SAVING,
      payload: { message: savingMsg },
    });

    expect(newState.meta.error).toBe(false);
    expect(newState.meta.loading).toBe(false);
    expect(newState.meta.saving).toBe(savingMsg);
    expect(newState).toMatchSnapshot();
  });

  it('Sets up saved state correctly', () => {
    const newState = samplesReducer({
      ...oneSampleState,
      meta: {
        ...oneSampleState[sample1.uuid].meta,
        loading: false,
        saving: true,
        error: false,
      },
    }, { type: SAMPLES_SAVED });

    expect(newState.meta.error).toBe(false);
    expect(newState.meta.loading).toBe(false);
    expect(newState.meta.saving).toBe(false);
    expect(newState).toMatchSnapshot();
  });

  it('Stores error state correctly', () => {
    const errMsg = 'Error message';

    const newState = samplesReducer({
      ...oneSampleState,
      meta: {
        ...oneSampleState[sample1.uuid].meta,
        loading: false,
        saving: true,
        error: true,
      },
    }, {
      type: SAMPLES_ERROR,
      payload: {
        error: errMsg,
      },
    });

    expect(newState.meta.error).not.toBe(false);
    expect(newState.meta.error).toBe(errMsg);
    expect(newState.meta.loading).toBe(false);
    expect(newState.meta.saving).toBe(false);
    expect(newState).toMatchSnapshot();
  });

  it('Inserts sample metadata correctly', () => {
    const metadataKey = 'metadata-test';
    const metadataValue = 'value';

    const sampleWithMetadata = {
      ...oneSampleState,
      [oneSampleState[mockUuid1]]: {
        metadata: {},
      },
    };

    const newState = samplesReducer(sampleWithMetadata, {
      type: SAMPLES_UPDATE,
      payload: {
        sampleUuid: mockUuid1,
        sample: { metadata: { [metadataKey]: metadataValue } },
      },
    });

    expect(newState[mockUuid1].metadata[metadataKey]).toEqual(metadataValue);
    expect(newState).toMatchSnapshot();
  });

  it('Updates sample metadata correctly', () => {
    const metadataKey = 'metadata-test';
    const oldValue = 'old-value';
    const newValue = 'new-value';

    const sampleWithMetadata = {
      ...oneSampleState,
      [oneSampleState[mockUuid1]]: {
        metadata: {
          [metadataKey]: oldValue,
        },
      },
    };

    const newState = samplesReducer(sampleWithMetadata, {
      type: SAMPLES_UPDATE,
      payload: {
        sampleUuid: mockUuid1,
        sample: { metadata: { [metadataKey]: newValue } },
      },
    });

    expect(newState[mockUuid1].metadata[metadataKey]).toEqual(newValue);
    expect(newState).toMatchSnapshot();
  });

  it('Deletes sample metadata correctly', () => {
    const metadataKey = 'metadata-test';
    const metadataValue = 'old-value';

    const sampleWithMetadata = {
      ...oneSampleState,
      [oneSampleState[mockUuid1]]: {
        metadata: {
          [metadataKey]: metadataValue,
        },
      },
    };

    const newState = samplesReducer(sampleWithMetadata, {
      type: SAMPLES_METADATA_DELETE,
      payload: {
        sampleUuid: mockUuid1,
        metadataKey,
      },
    });

    expect(newState[mockUuid1].metadata[metadataKey]).toBeUndefined();
    expect(newState).toMatchSnapshot();
  });
});
