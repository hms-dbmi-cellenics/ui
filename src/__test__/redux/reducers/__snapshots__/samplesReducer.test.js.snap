// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`samplesReducer Adds a new sample correctly 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "test sample",
    "options": {},
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [],
  },
  "qwe234": {
    "complete": false,
    "createdDate": "2021-01-02T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-02T14:48:00.000Z",
    "metadata": {},
    "name": "test sample 2",
    "options": {},
    "type": null,
    "uuid": "qwe234",
  },
}
`;

exports[`samplesReducer Adds validating to an experiment 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "test sample",
    "options": {
      "someOption": false,
    },
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [
      "validatingExperimentId",
    ],
  },
  "qwe234": {
    "complete": false,
    "createdDate": "2021-01-02T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-02T14:48:00.000Z",
    "metadata": {},
    "name": "test sample 2",
    "options": {
      "someOption": false,
    },
    "type": null,
    "uuid": "qwe234",
  },
}
`;

exports[`samplesReducer Delete samples correctly 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "test sample",
    "options": {},
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [],
  },
}
`;

exports[`samplesReducer Deletes sample metadata correctly 1`] = `
{
  "[object Object]": {
    "metadata": {
      "metadata-test": "old-value",
    },
  },
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "test sample",
    "options": {},
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [],
  },
}
`;

exports[`samplesReducer Handles experimentMetadataRename correctly 1`] = `
{
  "asd123": {
    "experimentId": "mockExpId",
    "metadata": {
      "metadata-new": "value",
    },
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [],
  },
}
`;

exports[`samplesReducer Handles samplesValueInMetadataTrackUpdated correctly 1`] = `
{
  "asd123": {
    "metadata": {
      "metadata-test": "new-value",
    },
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [],
  },
}
`;

exports[`samplesReducer Inserts a new sample correctly 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "test sample",
    "options": {},
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [],
  },
}
`;

exports[`samplesReducer Inserts sample metadata correctly 1`] = `
{
  "[object Object]": {
    "metadata": {},
  },
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {
      "metadata-test": "value",
    },
    "name": "test sample",
    "options": {},
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [],
  },
}
`;

exports[`samplesReducer Loads samples correctly 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "test sample",
    "options": {},
    "type": null,
    "uuid": "asd123",
  },
  "ids": [
    "asd123",
    "qwe234",
  ],
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [],
  },
  "qwe234": {
    "complete": false,
    "createdDate": "2021-01-02T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-02T14:48:00.000Z",
    "metadata": {},
    "name": "test sample 2",
    "options": {},
    "type": null,
    "uuid": "qwe234",
  },
}
`;

exports[`samplesReducer Removes validating from an experiment 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "test sample",
    "options": {
      "someOption": false,
    },
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "validating": [
      "otherValidatingExperimentId",
    ],
  },
  "qwe234": {
    "complete": false,
    "createdDate": "2021-01-02T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-02T14:48:00.000Z",
    "metadata": {},
    "name": "test sample 2",
    "options": {
      "someOption": false,
    },
    "type": null,
    "uuid": "qwe234",
  },
}
`;

exports[`samplesReducer Sets up saved state correctly 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "test sample",
    "options": {},
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
  },
}
`;

exports[`samplesReducer Sets up saving state correctly 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "test sample",
    "options": {},
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": "Saving",
  },
}
`;

exports[`samplesReducer Stores error state correctly 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "test sample",
    "options": {},
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": "Error message",
    "loading": false,
    "saving": false,
  },
}
`;

exports[`samplesReducer Updates a sample correctly 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "updated name",
    "options": {},
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [],
  },
}
`;

exports[`samplesReducer Updates options correctly 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-01T14:48:00.000Z",
    "metadata": {},
    "name": "test sample",
    "options": {
      "someOption": true,
    },
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [],
  },
  "qwe234": {
    "complete": false,
    "createdDate": "2021-01-02T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {},
    "lastModified": "2021-01-02T14:48:00.000Z",
    "metadata": {},
    "name": "test sample 2",
    "options": {
      "someOption": true,
    },
    "type": null,
    "uuid": "qwe234",
  },
}
`;

exports[`samplesReducer Updates sample files correctly 1`] = `
{
  "asd123": {
    "complete": false,
    "createdDate": "2021-01-01T14:48:00.000Z",
    "error": false,
    "experimentId": null,
    "files": {
      "features10x": {
        "error": false,
        "objectKey": "",
        "size": 0,
        "success": false,
        "upload": {
          "status": null,
        },
      },
    },
    "lastModified": "newLastModified",
    "metadata": {},
    "name": "test sample",
    "options": {},
    "type": null,
    "uuid": "asd123",
  },
  "meta": {
    "error": false,
    "loading": false,
    "saving": false,
    "validating": [],
  },
}
`;
